### ISSUE TITLE
add xkcd style

### ISSUE THREAD

**Issue by WeatherGod** [matplotlib/matplotlib#5992]
Can't believe we don't have this one already. Not perfectly trivial, as we would need a way to specify the special patheffect from the rcfile, but, essentially, this would replace `plt.xkcd()`.

Should also look into finding an open font that we can package that matches Humor Sans.

**Comment by DadAtH-me** [comment]
[Comic Neue](http://comicneue.com/) should fit that requirement.

I've been trying to figure out how to get it working on FreeBSD (and failing).

**Comment by souravsingh** [comment]
I would like to work on the issue.How do I start?

**Comment by jenshnielsen** [comment]
@souravsingh Awesome

I suggest having a look at the existing styles (.sty) files in https://github.com/matplotlib/matplotlib/tree/master/lib/matplotlib/mpl-data/stylelib to see how they work.

The code that defines the pyplot.xkcd is here: https://github.com/matplotlib/matplotlib/blob/master/lib/matplotlib/pyplot.py#L353 so as a first step you can copy the rc params changes from there into a .sty file.

**Comment by PoldoSbaffini** [comment]
@WeatherGod 
> Should also look into finding an open font that we can package that matches Humor Sans

No, Humor Sans is released under the "SIL Open Font License, Version 1.1"

**Comment by rcomer** [comment]
This still seems like something we should have.

**Comment by anntzer** [comment]
@rcomer Perhaps we can decide whether we want to move forward with #14943 (or something along these lines) -- I don't really want to spend time arguing for it anymore.  If we really don't want python style files, even restricted to matplotlib's own styles, then I think we should close this as effectively unfixable (because instantiating patheffects, even declaratively, necessarily means importing a python class and thus running arbitrary python code).

**Comment by timhoffm** [comment]
What could speak against python style files:

- security: Users can inject arbitrary code. - IMHO there are various other ways to do this. (If it's really of concern, one could even guard this behind a flag)
- API: With great power comes great responsibility. For public python style files we would need good instructions how to handle them.

Both aspects do not apply as long as we keep python style files internal. - I think this is a good way forward a) to provide `xkcd` as style and b) to learn how python style files could be used without locking in on a certain API yet. Thus I support going forward with interal python style files. We can later decide whether we want to make this public as well.

**Comment by anntzer** [comment]
OK, I rebased and reopened #14943, let's discuss that over there.
